articles/building-an-appointment-scheduling-telegram-bot/index.md

```diff
@@ -0,0 +1,514 @@
+
+Next, you need to create a dispatch handler for the `/start` command. A command in telegram is any text that starts with `/`.
+
+You can also add other handlers such as image handler, text handler, regex handler, and many more. We are starting with a command handler because that is the first thing every user enters into a Telegram Bot.
+
+Copy and paste the code below at the end of your python file to call the dispatch handler to connect a command handler to the “start” command. The “start” command triggers the `start` method. The updater checks for messages from the user.
+
+```python
+
+dispatcher.add_handler(CommandHandler("start", start))
+
+updater.start_polling()
+
+```
+
+After updating your python file, run with the code below:
+
+
+```bash
+
+# python file name is app.py
+
+python app.py
+
+```
+While your python app is running, open your telegram app and search for the bot you created.
+
+![search_bot](/engineering-education/building-an-appointment-scheduling-telegram-bot/search_bot.png)
+
+Open your bot and start it by typing in the `/start` command. You will now receive a welcome message from the bot, as seen in the image below:
+
+![start_bot](/engineering-education/building-an-appointment-scheduling-telegram-bot/start_bot.png)
+
+#### Creating New Appointments
+
+To enable your telegram bot to create new appointments in the fauna database, copy and paste the code below in your python file.
+
+```python
+
+def add_appointment(update, context):
+
+    chat_id = update.effective_chat.id
+
+
+    user = client.query(q.get(q.match(q.index("users_index"), chat_id)))
+
+    client.query(q.update(q.ref(q.collection("Users"), user["ref"].id()), {"data": {"last_command": "add_appointment"}}))
+
+    context.bot.send_message(chat_id=chat_id, text="Enter the appointment event you want to add along with its due in this format(mm/dd/yyyy) date separated by a comma 😁")
+
+```
+
+In the code above, you defined a method called `add_appointment` where you passed `update` and `context` as parameters. In the function, you saved the user’s message and `chat_id`. You then made use of the `chat_id` to query the FQL client using  the `users_index` index to check the user data that matches the `chat_id`.
+
+Next, you updated the `last_command` field in the user's data to save the `add_appointment` command the user sent. You then sent a message requesting the user to enter the name of the appointment and its due date separated by a comma.
+
+You have to create another method called `echo` that will collect the name and the due date for the new appointment then save it to the database. Copy and paste the code below after your `add_appointment` method in your python file.
```

Comment:
Now you have to create another method called `echo` that will collect the name and the due date for the new appointment then save it to the database.